erpnext/assets/doctype/asset/asset.py

```diff
@@ -453,7 +453,7 @@ def make_gl_entries(self):
 
 		if ((self.purchase_receipt \
 			or (self.purchase_invoice and frappe.db.get_value('Purchase Invoice', self.purchase_invoice, 'update_stock')))
-			and self.purchase_receipt_amount and self.available_for_use_date <= nowdate()):
+			and self.purchase_receipt_amount and getdate(self.available_for_use_date) <= nowdate()):
```

Comment:
this won't fix it, the datatype for `available_for_use_date` is already set to `Date`.

`nowdate()` returns a string, which needs to be converted to datetime using `getdate()`. so the correct fix for this issue would be:

```suggestion
			and self.purchase_receipt_amount and self.available_for_use_date <= getdate(nowdate())):
```

Reply:
Else where in the same code getdate is used for available_for_use_date, hence thought so. This is a background job, to be verified. Thank you